store/tikv/client.go

```diff
@@ -48,6 +49,67 @@ type Client interface {
 	SendReq(ctx goctx.Context, addr string, req *tikvrpc.Request) (*tikvrpc.Response, error)
 }
 
+type connArray struct {
+	lock  *sync.Mutex
+	index uint32
+	v     []*grpc.ClientConn
+}
+
+func newConnArray(maxSize uint32, addr string) (*connArray, error) {
+	a := &connArray{
+		lock:  &sync.Mutex{},
+		index: 0,
+		v:     make([]*grpc.ClientConn, maxSize),
+	}
+	if err := a.Init(addr); err != nil {
+		return nil, err
+	}
+	return a, nil
+}
+
+func (a *connArray) Init(addr string) error {
+	for i := range a.v {
+		conn, err := grpc.Dial(
+			addr,
+			grpc.WithInsecure(),
+			grpc.WithTimeout(dialTimeout),
+			grpc.WithInitialWindowSize(grpcInitialWindowSize),
+			grpc.WithInitialConnWindowSize(grpcInitialConnWindowSize),
+			grpc.WithUnaryInterceptor(grpc_prometheus.UnaryClientInterceptor),
+			grpc.WithStreamInterceptor(grpc_prometheus.StreamClientInterceptor))
+		if err != nil {
+			// Cleanup if the initialization fails.
+			a.doClose()
+			return errors.Trace(err)
+		}
+		a.v[i] = conn
+	}
+	return nil
+}
+
+func (a *connArray) Get() *grpc.ClientConn {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+	conn := a.v[a.index]
+	a.index = (a.index + 1) % uint32(len(a.v))
```

Comment:
It's not necessary since the whole connection map in `rpcClient` is protected by `rpcClient.isClosed`.